Port to Plasma 6

RiskoZS opened the pull request:
This PR updates the plugin to Plasma 6 and cleans up the CMake and README files a bit.

Notable side effects:
* Support for Plasma 5 has been removed. As there are people still using Plasma 5, I recommend merging this PR into a new branch instead of `master` (or archiving the current `master` into a new branch).
* The plugin ID, and thus the configuration group name, has changed from `Translator` to `krunner_translator`; users who upgraded from Plasma 5 will need to reconfigure the plugin.
* The installation instructions have been updated; the README lists the commands to run ~~and the installation/uninstallation scripts have been removed. This is a somewhat opinionated change, as I believe the scripts did not offer much value and I'd prefer users read the commands they run on their machine.~~
* The uninstallation scripts now attempt to install the required build and runtime dependencies (including `translate-shell`) via `apt` or `dnf`.
* ~~The README no longer lists dependencies for OpenSUSE.~~

Closes https://github.com/naraesk/krunner-translator/issues/30.

thegrasshopper104 commented:
I tried simply installing your version from the store but it fails:
```
/home/aronkvh/.local/share/krunner-sources/krunner-translator-1.5.0/install.sh
/home/aronkvh/.local/share/krunner-sources/krunner-translator-1.5.0/install.sh: sor: 8: kf5-config: not found
/home/aronkvh/.local/share/krunner-sources/krunner-translator-1.5.0/install.sh: sor: 8: kf5-config:  not found
/home/aronkvh/.local/share/krunner-sources/krunner-translator-1.5.0/install.sh: sor: 8: cmake:  not found
```

RiskoZoSlovenska commented:
Hi @thegrasshopper104, I haven't published my fork to the KDE store; you tried to install the original version. I forgot to remove the store link from my repo and will do so now.

The plugin uses `translate-shell` for TTS, which in turn needs `mpv`, `mplayer` or `mpg123`. Make sure you have one of those installed. Thanks for your feedback!

Thaodan commented on uninstall.sh:
```diff
@@ -1,9 +1,4 @@
-kquitapp5 krunner 2> /dev/null; kstart5 --windowclass krunner krunner > /dev/null 2>&1 &
+#!/bin/sh
 
+sudo cmake --build build --target uninstall
```
This change is good. Better that way the script isn't tied to make.

RiskoZS committed:
Port to Plasma 6
* Update to Plasma 6 APIs
* Fix some Plasma 6 errors/warnings
* Update CMakeLists.txt and README to get the correct dependencies
* Fix README reporting incorrect defaults
* Update build instructions
* Remove unnecessary files
* Possibly fix some configuration bugs
* Misc. formatting fixes

RiskoZS committed:
Re-add un/installation scripts
They now also attempt to install all the necessary dependencies
They might be helpful for users installing the plugin from the KDE store

RiskoZS committed:
Fix line breaks in the README
Who thought that trailing whitespace being significant was a good idea?

RiskoZS committed:
Re-add `set -e` to uninstall script

RiskoZoSlovenska commented on uninstall.sh:
```diff
@@ -1,9 +1,4 @@
-#!/bin/bash
-
-# Exit immediately if something fails
-set -e
```
Not sure, I think that may have been an oversight on my end; will fix

RiskoZoSlovenska commented on uninstall.sh:
```diff
@@ -1,9 +1,4 @@
-
-cd build
-sudo make uninstall
-kquitapp5 krunner 2> /dev/null; kstart5 --windowclass krunner krunner > /dev/null 2>&1 &
```
I am not sure there is a big difference; `killall` seems simpler and more portable to me as it's not tied to a specific KDE version.

Another alternative to consider is `systemctl --user restart plasma-krunner`; I think it is much cleaner, but it assumes that the user is using systemd and that krunner is set up to use that service name.

RiskoZoSlovenska commented:
I rebased on `master`.

From https://github.com/naraesk/krunner-translator/issues/30#issuecomment-3315988382:
> This changes look good more or less but I don't agree with advising users to install the plugin to /usr without the plugin being packaged.

I don't disagree, but I'm not sure there's an alternative. Unless I'm overlooking something, the Qt plugin dir (the directory in which krunner plugins live) is under `/usr/lib64/qt6/plugins`. Furthermore, this plugin installs to `/usr` even without this PR, and I haven't been able to find a different krunner plugin that installs itself elsewhere. Also note that this PR doesn't hardcode `/usr` anywhere; if the Qt plugin dir changes, I have no reason to believe the installation directory won't change accordingly.

> Similarly the readme changes are not an improvement IMHO.

Could you elaborate? Is there anything specific you don't like?

From https://github.com/naraesk/krunner-translator/issues/30#issuecomment-3315990533:
> Writing to /usr is impossible on fedora atomic unless it's a rpm package

I have plans to set up and PR a CI workflow that will build rpm and deb packages.

RiskoZS committed:
Improve dependencies and add openSUSE

* Support `zypper` in the install script
* Use `cmake(foo)` instead of `foo-devel` on Fedora and openSUSE
* Replace mentions of Ubuntu 24.10 (EOL) with 25.04
* Tweak display of dependencies in the README

RiskoZS committed:
Consistently use `killall -q` to restart krunner
`kquitapp6` mysteriously fails in some cases

RiskoZS committed:
Replace deprecated `QCheckBox::stateChanged`

Deprecated in Qt 6.9, replaced by `QCheckBox::checkStateChanged` since Qt 6.7

stone-w4tch3r commented:
Any plans to merge this? Any help needed?

naraesk commented:
Sorry for the delay. I've just finished porting the runner to Plasma 6 myself, having overlooked this pull request. Since I've already incorporated your changes into the install script and README, I'm going to close this merge request. Thanks for your contribution!
